### PR TITLE
Share more code between "Generate Constructor", "Add Parameter" and "Initialize Member from Parameter"

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Precursor to https://github.com/dotnet/roslyn/pull/76352

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@JoeRobich @ToddGrun ptal.

File: src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs [hunk 1]
```diff
@@ -103,12 +110,13 @@ public static async Task<Solution> AddParameterAsync(
             if (syntaxFacts is null)
                 continue;
 
+            var semanticDocument = await SemanticDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
```
Comment (ToddGrun) [inline src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs:113]: >semanticDocument [](http://example.com/codeflow?start=16&length=16)

used?

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs:113]: nope.  removing.

Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs:113]:
```suggestion
```

File: src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs [hunk 2]
```diff
@@ -11,14 +11,20 @@
...
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.AddParameter;
 
 internal static class AddParameterService
 {
+    private static readonly SyntaxAnnotation s_annotation = new();
+
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs:27]:
```suggestion
```

File: src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs [hunk 3]
```diff
@@ -60,12 +60,10 @@ protected internal sealed class State
         public ImmutableDictionary<string, string> ParameterToNewPropertyMap { get; private set; }
         public bool IsContainedInUnsafeType { get; private set; }
 
-        private State(TService service, SemanticDocument document, NamingRule fieldNamingRule, NamingRule propertyNamingRule, NamingRule parameterNamingRule)
+        private State(TService service, SemanticDocument document, NamingRule parameterNamingRule)
```
Comment (ToddGrun) [inline src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs:63]: >private State(TService service, SemanticDocument document, NamingRule parameterNamingRule) [](http://example.com/codeflow?start=8&length=90)

very unlike you not to change this to a primary ctor :)

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**